### PR TITLE
Added Graph QL Support

### LINKED ISSUE #117: Add GraphQL Support
**Tell us about the task you want to perform and are unable to do so because the feature is not available**
GraphQL, a query language for APIs and a runtime for executing those queries, has gained popularity due to its flexibility and efficiency in fetching data. Testing GraphQL APIs is critical for validating schema consistency, query correctness & error handling. By providing support for testing, visualization & integration code generation of GraphQL APIs, we can ensure their quality and reliability.

### PR DESCRIPTION
**Pull request by BrawlerXull**
## PR Description

Added GraphQL Support to the Application

## Related Issues

- Related Issue #117 
- Closes #117 

### Checklist
- [ ] I have gone through the [contributing guide](https://github.com/foss42/apidash/blob/main/CONTRIBUTING.md)
- [ ] I have run the tests (`flutter test`) and all tests are passing

## Added/updated tests?
_We encourage you to add relevant test cases._

- [ ] Yes
- [x] No, and this is why: Waiting for maintainers review so that I may add testcases
- [ ] I need help with writing tests

### CODE REVIEW

**Comment by BrawlerXull** [comment]
https://github.com/foss42/apidash/assets/111864031/93c1c87e-097d-46a3-9a89-34323d420630

@ashitaprasad @animator I've added graphql support this is the working video of the prototype there are many bugs yet to be fixed!
I wanted a review of weather my approach is correct or not?

File: lib/consts.dart [hunk 1]
```diff
@@ -255,6 +256,7 @@ const kMethodsWithBody = [
   HTTPVerb.put,
   HTTPVerb.patch,
   HTTPVerb.delete,
+  HTTPVerb.graphql,
```
Comment (mmjsmohit) [inline lib/consts.dart:259]: Same comment that I left in the code above. Instead, you can define a `ProtocolType` enumeration and include HTTP and GraphQL.

File: lib/services/http_service.dart
Comment (mmjsmohit) [inline lib/services/http_service.dart:1]: A separate service file must be made for GraphQL instead of modifying this one.

Comment (BrawlerXull) [reply lib/services/http_service.dart:1]: Thanks for giving the info!
I was also wondering the same :)
I wanted to ask if the logic of implementation
`String requestBody = '{"query": "${body!.replaceAll('\n', '')}"}';` is correct and the UI design is on the point 
Is there's a suggestion on that I would love to work on it and solve the bugs :)

**Comment by opxdelwin** [comment]
@BrawlerXull , The implementation looks good, but why did you put `GRAPHQL` as request type in url endpoint, as mentioned by @mmjsmohit ? 

Such a HTTP method doesn't exist. It'd be better if you could put `GRAPHQL` into body type, as implemented by [Postman](https://assets.postman.com/postman-docs/v10/graphql-http-variables-v10-22.jpg)

**Comment by BrawlerXull** [comment]
> Such

Yes! as mentioned earlier I was also thinking that adding it in HTTP method is incorrect 😄
 I just wanted to have a review on implementation so that I can move further with a create vision of implementation and the design of the application

Thanks for the review @opxdelwin @mmjsmohit

**Comment by BrawlerXull** [comment]
I would love to hear maintainer's take on this! 😄

**Comment by opxdelwin** [comment]
But wouldn't that take `GraphQL` side of requests into an entire separate side? It'd feel like QL developers are different than the rest... My preference would be one implemented by Postman as the image above. 
End of the day, GraphQL's body is different, using the same `POST` method, right?

**Comment by animator** [comment]
@BrawlerXull Please make this change https://github.com/foss42/apidash/issues/307

**Comment by animator** [comment]
Handling GraphQL API is different from REST HTTP API and the current approach needs a lot of rework and usage of graphQL packages.
Closing this PR. Please review the approach and raise a new PR.

**Pull request closed without merging**